src/connectStyle.js

```diff
@@ -175,10 +176,22 @@ export default (componentStyleName, componentStyle = {}, mapPropsToStyleNames, o
         return _.uniq(mapPropsToStyleNames(styleNames, props));
       }
 
+      isRefDefined() {
+        // Ref key exists in the props when it's defined on the component
+        // instance, but it has undefined value and it is non-enumerable.
+        // When it's not defined it doesn't exist in the props.
+        // This is abuse of the specifc ref behaviour but it optimizes number of created refs.
+        const propKeys = Object.getOwnPropertyNames(this.props);
+        return _.indexOf(propKeys, 'ref') >= 0;
+      }
+
       resolveAddedProps() {
         const addedProps = {};
         if (options.withRef) {
-          addedProps.ref = 'wrappedInstance';
+          console.warn('withRef is deprecated');
+        }
+        if (this.isRefDefined()) {
```

Comment:
Sorry for the long delay, @MrBr is on vacation at the moment.

Unfortunately I cannot merge this, because it will break the `setNativeProps` (see: https://github.com/shoutem/theme/blob/develop/src/connectStyle.js#L144). `setNativeProps` in `connectStyle` expects to have a ref to the underlying component. In order for everything to continue working as before we should:
1. capture refs for all class components
2. update `setNativeProps` to throw a warning if the ref doesn't exist

Could you make those changes, so that we can finally merge this? You can detect a stateless component using something like this: https://stackoverflow.com/questions/41488713/react-how-to-determine-if-component-is-stateless-functional